### DISCUSSION TITLE
Interfacing PyTorch extensions with CUDA/C++ library using CMake

### DISCUSSION THREAD

**Discussion by woywoy123** [scikit-build/scikit-build#1004] [Show and tell]
Hello All, 

I thought this little project might be helpful for anyone exploring how to interface PyTorch's C++/CUDA API with Cmake and completely remove setuptools. The aim of this little project was to write a lot of commonly used functions in High Energy Particle Physics in Native CUDA/C++ tensor functions. Do keep in mind, the migration is currently a work in progress. But a complete list of functions can be found in the README. 

REPO: https://github.com/woywoy123/AnalysisG/tree/master/torch-extensions

**Comment by henryiii** [comment]
It's in "failed-installers". :( Is it not working? If so, why?

**Reply by woywoy123** [reply]
Ah this happens sometimes with the "Github actions". Usually due to a busy machine. I just need to restart it. I am just preparing to commit the code =)

**Reply by woywoy123** [reply]
Oh I see which directory you mean. This was related to Pytorch CUDA really not liking PyBind11. There is actually an active issue about this. I am circumventing this entirely by loading it with torch. Although I am trying to figure out a more elegant solution.
